GitCommands/Remote/GitRemoteController.cs

```diff
@@ -69,12 +68,12 @@ public class GitRemoteController : IGitRemoteController
         internal static readonly string DisabledSectionPrefix = "-";
         internal static readonly string SectionRemote = "remote";
         private static readonly object SyncRoot = new object();
-        private readonly IGitModule _module;
+        private readonly Func<IGitModule> _getModule;
 
 
-        public GitRemoteController(IGitModule module)
+        public GitRemoteController(Func<IGitModule> getModule)
         {
-            _module = module;
+            _getModule = getModule;
             Remotes = new BindingList<GitRemote>();
```

Comment:
It seems risky to have a state (Remotes) captured from module that can change overtime.

Reply:
this is needs to be refactored out, as it makes the controller stateful...
i'll do as a separate piece of work